packages/tree/src/Tree.tsx

```diff
@@ -0,0 +1,381 @@
+        item.isSelected = isSelected;
+        item.areAllChildrenSelected = isSelected;
+      });
+    }
+
+    setSelectedList(getSelectedItems(items));
+  };
+
+  const toggleSelectChildren = (item: TreeItem) => {
+    const isSelected = !areAllChildrenSelected(item);
+    item.isSelected = isSelected;
+    item.areAllChildrenSelected = isSelected;
+
+    for (const child of item.children || []) {
+      walkTree(child, (child) => {
+        child.isSelected = isSelected;
+        child.areAllChildrenSelected = isSelected;
+      });
+    }
+
+    setSelectedList(getSelectedItems(items));
+    setTreeItems([...treeItems]);
+  };
+
+  const onChildrenSelected = useCallback(() => {
+    setSelectedList(getSelectedItems(treeItems));
+  }, [getSelectedItems, treeItems]);
+
+  const onChildrenExpanded = () => {
+    setAllExpanded(areAllExpanded(items));
+    if (areAllExpanded(items)) {
+      setRootExpandAllText(COLLAPSE_ALL);
+    } else {
+      setRootExpandAllText(EXPAND_ALL);
+    }
+  };
+
+  if (!isReady) {
+    return null;
+  }
+
+  return (
+    <>
+      <div data-testid={`tree-view-${parentId || 'parent'}`} className="tree-view">
+        {isRoot && enableSearch && (
+          <div className="form-group">
+            <Label className="font-weight-bold">{searchLabel}</Label>
+            <Input
+              data-testid="tree-search-input"
+              type="text"
+              className="form-control"
+              id="search"
+              placeholder="Search"
+              onChange={(event: React.ChangeEvent<HTMLInputElement>) => search(event.target.value)}
+            />
+          </div>
+        )}
+
+        {isRoot && (
+          <div className="form-group mb-1">
+            <Button data-testid="btn-expand-all" id="btnExpandAll" color="link" className="p-0" onClick={toggleExpandAll}>
+              {rootExpandAllText}
+            </Button>
+          </div>
+        )}
+
+        <ul className="pl-0">
+          {treeItems.map((item) => (
+            <li data-testid={`tree-view-item-${item.id}`} key={`tree-view-item-${item.id}`}>
+              {!item.isHidden && (
+                <>
+                  <div className="d-flex">
+                    <FormGroup check>
+                      {!item.isDisabled && selectable && (
+                        <Input
+                          id={`chkSelect_${item.id}`}
+                          data-testid={`chk-tree-view-item-select-${item.id}`}
+                          type="checkbox"
+                          checked={item.isSelected}
+                          onChange={() => toggleSelect(item)}
+                        />
+                      )}
+                      <Label data-testid={`tree-view-item-${item.id}-label`} className={item.isDisabled ? 'text-muted' : ''} check>
+                        {item.name}
+                      </Label>
+                    </FormGroup>
+
+                    {item.children && item.children.length > 0 && (
```

Comment:
This can just be `item.children?. length > 0 &&`

Reply:
I agree, but this doesn't compile for some reason. 
![image](https://user-images.githubusercontent.com/8000861/171915207-5eb6a66d-de19-40ac-9163-b62fe784bc24.png)